internal/component/loki/source/syslog/types.go

```diff
@@ -42,6 +43,15 @@ func (sc *ListenerConfig) Validate() error {
 		return fmt.Errorf("syslog listener protocol should be either 'tcp' or 'udp', got %s", sc.ListenProtocol)
 	}
 
+	if sc.SyslogFormat != "" {
+		if sc.SyslogFormat != scrapeconfig.SyslogFormatRFC3164 && sc.SyslogFormat != scrapeconfig.SyslogFormatRFC5424 {
```

Comment:
Do you mind creating local RFC strings please?
```go
const (
	// A modern Syslog RFC
	SyslogFormatRFC5424 = "rfc5424"
	// A legacy Syslog RFC also known as BSD-syslog
	SyslogFormatRFC3164 = "rfc3164"
)
```

It would be good not to depend on Promtail for them, in case they get changed in the Promtail code.